### DISCUSSION TITLE
what are some good "seed" value? help on prompting with seed?

### DISCUSSION THREAD

**Discussion by hiqsociety** [ggml-org/llama.cpp#3203] [Q&A]
some seed makes a lot of repeating statement, some seed makes quick few response.
```
./main -m models/airoboros-l2-7b-2.2.Q6_K.gguf -ngl 35 -c 3680 -n 12288 -p "detailed minimum of 2400 words encyclopedia style markdown formatted article on 'how to generate passive income' in english with a sub section conclusion where appropriate is as follow:" -e
```
1. can someone advise how to make this prompt better?
2. -s 123 or 
./main -m models/airoboros-l2-7b-2.2.Q6_K.gguf -ngl 35 -c 3680 -n 12288 -p "detailed minimum of 2400 words encyclopedia style markdown formatted article on 'how to generate passive income' in english with a sub section conclusion where appropriate is as follow:" -e -s 123456

```
llm_load_tensors: offloading k cache to GPU
llm_load_tensors: offloaded 35/35 layers to GPU
llm_load_tensors: VRAM used: 7010 MB
...................................................................................................
llama_new_context_with_model: kv self size  = 1840.00 MB
llama_new_context_with_model: compute buffer total size =  255.47 MB
llama_new_context_with_model: VRAM scratch buffer: 254.00 MB

system_info: n_threads = 6 / 12 | AVX = 1 | AVX2 = 1 | AVX512 = 1 | AVX512_VBMI = 1 | AVX512_VNNI = 1 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | 
sampling: repeat_last_n = 64, repeat_penalty = 1.100000, presence_penalty = 0.000000, frequency_penalty = 0.000000, top_k = 40, tfs_z = 1.000000, top_p = 0.950000, typical_p = 1.000000, temp = 0.800000, mirostat = 0, mirostat_lr = 0.100000, mirostat_ent = 5.000000
generate: n_ctx = 3680, n_batch = 512, n_predict = 12288, n_keep = 0


 detailed minimum of 2400 words encyclopedia style markdown formatted article on 'how to generate passive income' in english with a sub section conclusion where appropriate is as follow:

# Table Of Contents
1. [Introduction](#introduction)
2. [What Is Passive Income?](#what-is-passive-income)
3. [Benefits of Passive Income](#benefits-of-passive-income)
4. [The Different Types Of Passive Income](#the-different-types-of-passive-income)
5. [Best Ways To Generate Passive Income](#best-ways-to-generate-passive-income)
    - 1. Buy And Hold Real Estate Investments
    - 2. Rental Property Investment
    - 3. Dividend Stocks (DRIPs)
    - 4. Selling Online Courses Or Digital Products
    - 5. Affiliate Marketing
    - 6. Start A Blog And Build An Audience
    - 7. Create An App
    - 8. Crowdfunding Platforms For Investment Opportunities
    - 9. Selling Stock Photography Or Graphic Design
    - 10. Peer-To-Peer Lending And Microfinancing Services
6. [Conclusion](#conclusion)
 [end of text]

llama_print_timings:        load time =   902.65 ms
llama_print_timings:      sample time =   107.20 ms /   285 runs   (    0.38 ms per token,  2658.48 tokens per second)
llama_print_timings: prompt eval time =    89.33 ms /    41 tokens (    2.18 ms per token,   458.99 tokens per second)
llama_print_timings:        eval time =  7303.62 ms /   284 runs   (   25.72 ms per token,    38.88 tokens per second)
llama_print_timings:       total time =  7649.33 ms
Log end
```

**Comment by staviq** [comment]
Seed is pretty much insignificant in this situation. In fact, if you look at `simple` example, you'll notice it doesn't even use seed, or randomness at all, and yet it still works.

There are no magic tricks here, the ability of a model, to follow your prompt exactly, depend entirely on the model itself and it's training. And the reality is, current generation of models is far from perfect.

Considering the prompt you are using, my personal subjective guesstimate, is that we will have to wait couple more years, I'd say at least 3, before local LLMs will be able to do the things you are asking from it.

**Comment by KerfuffleV2** [comment]
You should look up the model you're trying to use and its expected prompt format. I don't know about the 7b but the larger Airoboros models have a specific prompt format you should use to get good results. Like staviq said, the seed doesn't really matter. It's just an arbitrary number that will (possibly) give you different results. One seed isn't really better or worse than any other, the prompt is the important thing here and sampler settings.

From what I know, small models tend to be more sensitive to the prompt format. And, of course, they aren't as "smart" as bigger models either so you should keep your expectations realistic. Even models 10 times larger still don't come close to stuff like ChatGPT.
